packages/clay-label/src/index.tsx

```diff
@@ -101,51 +93,93 @@ const ClayLabelBase = React.forwardRef<HTMLSpanElement, IBaseProps>(
 			<span
 				{...otherProps}
 				className={classNames('label', className, {
-					'label-dismissible': closeButtonProps,
+					'label-dismissible': dismissible,
 					'label-lg': large,
 					[`label-${displayType}`]: displayType,
 				})}
 				ref={ref}
 			>
 				{children}
-
-				{closeButtonProps && (
-					<ClayLabelItemAfter>
-						<button
-							{...closeButtonProps}
-							className={classNames(
-								closeButtonProps.className,
-								'close'
-							)}
-							type="button"
-						>
-							<ClayIcon spritemap={spritemap} symbol="times" />
-						</button>
-					</ClayLabelItemAfter>
-				)}
 			</span>
 		);
 	}
 );
 
 interface IProps extends IBaseProps {
+	/**
+	 * HTML properties that are applied to the 'x' button.
+	 */
+	closeButtonProps?: React.ButtonHTMLAttributes<HTMLButtonElement> & {
+		ref?: (instance: HTMLButtonElement | null) => void;
+	};
+
+	/**
+	 * Pros to add to the inner label item
+	 */
 	innerElementProps?: React.ComponentProps<typeof ClayLabelItemExpand>;
+
+	/**
+	 * Path to the location of the spritemap resource used for Icon.
+	 */
+	spritemap?: string;
+
+	/**
+	 * Flag to indicate if component should include the close button
+	 */
+	withClose?: boolean;
 }
 
-const ClayLabel = React.forwardRef<HTMLAnchorElement | HTMLSpanElement, IProps>(
-	({children, href, innerElementProps = {}, ...otherProps}: IProps, ref) => {
+const ClayLabelHybrid = React.forwardRef<
+	HTMLAnchorElement | HTMLSpanElement,
+	IProps
+>(
+	(
+		{
+			children,
+			closeButtonProps,
+			href,
+			innerElementProps = {},
+			withClose = true,
+			spritemap,
+			...otherProps
+		}: IProps,
+		ref
+	) => {
 		return (
-			<ClayLabelBase {...otherProps} ref={ref}>
-				<ClayLabelItemExpand {...innerElementProps} href={href}>
-					{children}
-				</ClayLabelItemExpand>
-			</ClayLabelBase>
+			<ClayLabel dismissible={withClose} {...otherProps} ref={ref}>
+				{!withClose && children}
+
+				{withClose && (
+					<>
+						<ClayLabelItemExpand {...innerElementProps} href={href}>
+							{children}
+						</ClayLabelItemExpand>
+
+						{closeButtonProps && (
+							<ClayLabelItemAfter>
```

Comment:
It seems like this implementation doesn't consume `ClayLabelItemBefore` or am I wrong?

Reply:
Correct. Rather than add a new prop to the API, I went the route of breaking up Label into its low-level components so that users have more flexibility in how they want to use it. You can see the example of how we would add a before item at https://github.com/liferay/clay/pull/3081/files#diff-d25f4565a23c4c3cc5475dedd79d9ad1R62-R67

Reply:
Okay I see, good to know! Thanks